### PR TITLE
Admission controller for egress firewall

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
**- What this PR does and why is it needed**
This adds an admission controller to ovn-kubernetes, which enforces the one egressfirewall per namespace rule. Allowing for the removal of the validation of the names field in the CRD. 

**- Special notes for reviewers**
based on the admission controller work done by !Alexander Constantinescu
https://github.com/ovn-org/ovn-kubernetes/pull/1511

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: dist/images/gen_admission_keys.sh [hunk 1]
```diff
@@ -1,9 +1,15 @@
 #!/bin/bash
 
 # Generate the CA cert and private key
-openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=OVN-Kubernetes Admission Controller Webhook CA"
+       openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=OVN-Kubernetes Admission Controller Webhook CA"
```
Comment (danwinship) [inline dist/images/gen_admission_keys.sh:4]: gratuitous indentation?

also, if this PR is going to merge before Alexander's original one, then you should probably squash the fixes into the original commits

File: contrib/kind.sh [hunk 2]
```diff
@@ -483,7 +483,7 @@ create_ovn_kube_manifests() {
     --ovn-loglevel-nbctld="${OVN_LOG_LEVEL_NBCTLD}" \
     --egress-ip-enable=true \
     --v4-join-subnet="${JOIN_SUBNET_IPV4}" \
-    --v6-join-subnet="${JOIN_SUBNET_IPV6}"
+    --v6-join-subnet="${JOIN_SUBNET_IPV6}" \
```
Comment (danwinship) [inline contrib/kind.sh:486]: likewise, squash these fixes

File: go-controller/hack/update-codegen.sh [hunk 3]
```diff
@@ -56,12 +56,7 @@ echo "Generating CRDs"
 mkdir -p _output/crds
 controller-gen crd:crdVersions="v1"  paths=./pkg/crd/... output:crd:dir=_output/crds
 echo "Editing egressFirewall CRD"
-## We desire that only egressFirewalls with the name "default" are accepted by the apiserver. The only
-## way that we can put a pattern for validation on the name of the object which is embedded in
-## metav1.ObjectMeta it is required that we add it after the generation of the CRD.
-sed -i -e':begin;$!N;s/.*metadata:\n.*type: object/&\n            properties:\n              name:\n                type: string\n                pattern: ^default$/;P;D' \
-	_output/crds/k8s.ovn.org_egressfirewalls.yaml
-## It is also required that we restrict the number of properties on the 'to' section of the egressfirewall
+## tt is required that we restrict the number of properties on the 'to' section of the egressfirewall
```
Comment (danwinship) [inline go-controller/hack/update-codegen.sh:59]: "It" got changed to "tt" at the start there

File: go-controller/pkg/util/kube.go [hunk 4]
```diff
@@ -101,6 +101,7 @@ func newKubernetesRestConfig(conf *config.KubernetesConfig) (*rest.Config, error
 // NewKubernetesClientset creates a Kubernetes clientset from a KubernetesConfig
 func NewKubernetesClientset(conf *config.KubernetesConfig) (*kubernetes.Clientset, error) {
 	kconfig, err := newKubernetesRestConfig(conf)
+	klog.Errorf("KEYWORD: %+v\n", kconfig)
```
Comment (danwinship) [inline go-controller/pkg/util/kube.go:104]: leftover debugging?

**Review by danwinship — COMMENTED** [review]
this looks basically good, I think...

File: go-controller/hack/update-codegen.sh (hunk 3 above)
Comment (JacobTanenbaum) [reply go-controller/hack/update-codegen.sh]: done

File: go-controller/pkg/util/kube.go (hunk 4 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/util/kube.go]: removed

**Comment by JacobTanenbaum** [comment]
issue is stale my reopen later

**Pull request closed without merging**